sdk/trace/attributesmap.go

```diff
@@ -62,21 +61,23 @@ func (am *attributesMap) add(kv label.KeyValue) {
 	}
 }
 
-func (am *attributesMap) toSpanData(sd *trace.SpanData) {
-	len := am.evictList.Len()
-	if len == 0 {
-		return
+// toKeyValue copies the attributesMap into a slice of label.KeyValue and
+// returns it. If the map is empty, a zero-length slice is returned.
+// TODO: Is it more efficient to return a pointer to the slice?
+func (am *attributesMap) toKeyValue() []label.KeyValue {
+	var attributes []label.KeyValue
+
+	if am.evictList.Len() == 0 {
+		return attributes
 	}
 
-	attributes := make([]label.KeyValue, 0, len)
```

Comment:
Should this pre-allocation of capacity for the `attributes` slice be retained?  We already know how large it is going to be and can potentially save some reallocation.

Reply:
Makes sense to me. Fixed.
For my own understanding, does this affect performance in a meaningful way?

Reply:
I've added a benchmark for this function and pre-allocating like you've suggested is about 50% faster. Great catch!

Reply:
Without pre-allocation:

```
go test ./trace -bench BenchmarkAttributesMapToKeyValue
goos: linux
goarch: amd64
pkg: go.opentelemetry.io/otel/sdk/trace
BenchmarkAttributesMapToKeyValue-8        224655              4562 ns/op
PASS
ok      go.opentelemetry.io/otel/sdk/trace      1.231s
```

With pre-allocation:

```
go test ./trace -bench BenchmarkAttributesMapToKeyValue
goos: linux
goarch: amd64
pkg: go.opentelemetry.io/otel/sdk/trace
BenchmarkAttributesMapToKeyValue-8        534184              2142 ns/op
PASS
ok      go.opentelemetry.io/otel/sdk/trace      1.321s
```